Libs/MRML/Core/vtkMRMLSegmentationDisplayNode.cxx

```diff
@@ -365,62 +368,121 @@ void vtkMRMLSegmentationDisplayNode::SetSegmentDisplayProperties(std::string seg
       }
     }
 
-  // Save cached value of color
-  // TODO: remove this when terminology infrastructure is in place
+  if (modified)
+    {
+    this->Modified();
+    }
+}
+
+//---------------------------------------------------------------------------
+vtkVector3d vtkMRMLSegmentationDisplayNode::GetSegmentColor(std::string segmentID)
+{
+  double color[3] = {vtkSegment::SEGMENT_COLOR_INVALID[0], vtkSegment::SEGMENT_COLOR_INVALID[1], vtkSegment::SEGMENT_COLOR_INVALID[2]};
+  this->GetSegmentColor(segmentID, color);
+
+  vtkVector3d colorVtk(color[0], color[1], color[2]);
+  return colorVtk;
+}
+
+//---------------------------------------------------------------------------
+void vtkMRMLSegmentationDisplayNode::GetSegmentColor(std::string segmentID, double* color)
+{
+  if (!color)
+    {
+    vtkErrorMacro("GetSegmentColor: Invalid output color array");
+    return;
+    }
+
+  // Invalidate color
+  color[0] = vtkSegment::SEGMENT_COLOR_INVALID[0];
+  color[1] = vtkSegment::SEGMENT_COLOR_INVALID[1];
+  color[2] = vtkSegment::SEGMENT_COLOR_INVALID[2];
+
+  // Get segment
   vtkMRMLSegmentationNode* segmentationNode = vtkMRMLSegmentationNode::SafeDownCast(this->GetDisplayableNode());
-  if (segmentationNode)
+  if (!segmentationNode)
     {
-    vtkSegmentation* segmentation = segmentationNode->GetSegmentation();
-    vtkSegment* segment = segmentation->GetSegment(segmentId);
-    if (segment)
-      {
-      segment->SetDefaultColorWithoutModifiedEvent(properties.Color);
-      }
+    vtkErrorMacro("GetSegmentColor: No segmentation node associated to this display node");
+    return;
+    }
+  vtkSegmentation* segmentation = segmentationNode->GetSegmentation();
```

Comment:
For efficient implementation of segmentation sequences (allowing sequence nodes changing in time, replayed with reasonable speed) I had to allow setting Segmentation member from outside and so now it is possible that temporarily Segmentation is NULL. So, it might worth checking if segmentation is NULL.